### ISSUE TITLE
Bug in definition of wl and wr terms

### ISSUE THREAD

**Issue by ScottLemon** [gerlero/PCHIPInterpolation.jl#31]
I believe there may be an bug in the following lines:
https://github.com/gerlero/PCHIPInterpolation.jl/blob/cfc42876c2d14fb5a4c6a5e6b4a1d9144c02c7a4/src/PCHIPInterpolation.jl#L39-L40

Based on the scipy/Matlab implementations of the derivative function the wl and wr terms should be swapped so that:
```
wl = hl + 2hr
wr = 2hl + hr
```

See https://github.com/scipy/scipy/blob/main/scipy/interpolate/_cubic.py#L285 for the corresponding scipy version.

Due to this issue, the current Julia version gives subtly different results to both Matlab and scipy.

For example, the Python scipy version:
```
import numpy as np
import scipy.interpolate

xs = np.array([0.0,  1.2,  2.0,  5.0, 10.0, 11.0])
ys = np.array([2.0,  2.1,  1.0,  0.0,  0.0,  3.0])
yp = scipy.interpolate.pchip_interpolate(xs, ys, 2.5)
```

gives yp=0.714763623953731

The Matlab version:
```
xs = [0.0,  1.2,  2.0,  5.0, 10.0, 11.0]
ys = [2.0,  2.1,  1.0,  0.0,  0.0,  3.0]
yp = pchip(x, y, 2.5)
```
also gives yp=0.714763623953731

The current Julia version:
```
xs = [0.0,  1.2,  2.0,  5.0, 10.0, 11.0]
ys = [2.0,  2.1,  1.0,  0.0,  0.0,  3.0]
itp = Interpolator(xs, ys)
yp = itp(2.5)
```
gives yp=0.7592273552265896

**Comment by gerlero** [comment]
@ScottLemon interesting! I see that SciPy not only does it differently but also [notes the correct formula](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.PchipInterpolator.html), so this is a bug.

Would you submit a pull request with the required change (+ the test)?

**Comment by ScottLemon** [comment]
I have created a pull request that yields interpolated values that are now consistent with SciPy.

I added a test that simply checks if the interpolated values at the interval midpoints of the sample function are the same as those computed using SciPy. I imagine there is probably a better/more rigorous way to actually check the correctness of the implementation.

I looked through SciPy's [tests ](https://github.com/scipy/scipy/blob/9aee554b9f87d53eb32fd2b0318c45439c9d9950/scipy/interpolate/tests/test_polyint.py#L398)for their PCHIP implementation and it turns out that they had the same bug in their implementation that was [fixed back in 2015](https://github.com/scipy/scipy/commit/b127884e79ef5f10e0ba6d0ec58ebd6a799a8878). They use [an example from the NAG C](https://github.com/scipy/scipy/blob/9aee554b9f87d53eb32fd2b0318c45439c9d9950/scipy/interpolate/tests/test_polyint.py#L439) implementation as a smoke test for the derivative calculation.

**Comment by gerlero** [comment]
> I have created a pull request that yields interpolated values that are now consistent with SciPy.

Thanks & merged. I'll create a new release ASAP.

> I added a test that simply checks if the interpolated values at the interval midpoints of the sample function are the same as those computed using SciPy. I imagine there is probably a better/more rigorous way to actually check the correctness of the implementation.

I think the way you're doing it is just fine for our purposes.

> I looked through SciPy's [tests ](https://github.com/scipy/scipy/blob/9aee554b9f87d53eb32fd2b0318c45439c9d9950/scipy/interpolate/tests/test_polyint.py#L398)for their PCHIP implementation and it turns out that they had the same bug in their implementation that was [fixed back in 2015](https://github.com/scipy/scipy/commit/b127884e79ef5f10e0ba6d0ec58ebd6a799a8878). They use [an example from the NAG C](https://github.com/scipy/scipy/blob/9aee554b9f87d53eb32fd2b0318c45439c9d9950/scipy/interpolate/tests/test_polyint.py#L439) implementation as a smoke test for the derivative calculation.

Interesting!

**Comment by gerlero** [comment]
Fixed in v0.1.6. Thanks @ScottLemon!
